### PR TITLE
feat: add instructions for using df examples [MONET-1563]

### PR DESCRIPTION
**Pull request by kdamball**
## Purpose

Instructions file on how to develop delivery functions 

## Approach

Adding instructions markdown file instead of everything in README.

### CODE REVIEW

File: examples/hosted-delivery-function-potterdb/INSTRUCTIONS.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## Building your Delivery Function
+
+### Creating an app
+You can create an app using CLI using `npm run create-app-definition`. This will prompt you to enter details for your new app and organization details. Make sure your app is enabled for `Entry field` option with both Short text and JSON Object selected in the supported field types.
+
+Alternatively, you can set your app and organization details by setting them as environment variables.
```
Comment (BasKiers) [inline examples/hosted-delivery-function-potterdb/INSTRUCTIONS.md:6]: Personally I would find it logical to mention that they could alternatively use the web-app to create a new link (with short description where they should look). The part about environment variables is listed under the next header

File: examples/hosted-delivery-function-potterdb/INSTRUCTIONS.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+### Using environment variables
+You can also set the following environment variables as listed below:
```
Comment (BasKiers) [inline examples/hosted-delivery-function-potterdb/INSTRUCTIONS.md:9]:
```suggestion
You will need to set the following environment variables as listed below:
```

File: examples/hosted-delivery-function-potterdb/INSTRUCTIONS.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+- `CONTENTFUL_ACCESS_TOKEN`: User management token used for authentication/authorization
+- `CONTENTFUL_ORG_ID`: Organization id where the app lives under
+- `CONTENTFUL_APP_DEF_ID`: App definition id for identifying the app
+
+### Uploading the code to Contentful
+It as simple using the CLI command `npm run upload-ci`. This will perform two actions: upload the code, linking it to the app, and then finally activating the code ready for usage in both 
+
+### Assigning an app to a field
+Delivery functions are meant to help with resolving field data, meaning the app has to be assigned to a field location. You can read more about different app locations [here](https://www.contentful.com/developers/docs/extensibility/app-framework/locations/). The field is also required to be of the supported types for delivery function: Short text (`Symbol`) or JSON fields (`Object`). If these requirements are met, the UI has an option to enable resolution for the content type's fields during delivery when using GraphQL.
```
Comment (BasKiers) [inline examples/hosted-delivery-function-potterdb/INSTRUCTIONS.md:18]:
```suggestion
Delivery functions are meant to help with resolving field data, meaning the app has to be assigned to a field location. You can read more about different app locations [here](https://www.contentful.com/developers/docs/extensibility/app-framework/locations/). 

The field is required to be one of the supported types for delivery function: Short text (`Symbol`) or JSON fields (`Object`). If these requirements are met, the UI has an option to enable resolution for the content type's fields in the GraphQL API.
```

File: examples/hosted-delivery-function-potterdb/INSTRUCTIONS.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+To enable both options you can also use the [web UI](https://app.contentful.com/deeplink?link=app-definition-list) by going under the "Locations" section in the app details UI, then selecting the appropriate supported locations. After [installing an app to your space environment](https://www.contentful.com/developers/docs/extensibility/app-framework/tutorial/#install-your-app-to-a-space), you can go under the [respective content type](https://app.contentful.com/deeplink?link=content-model) you want to assign the app and visit the appearance section for supported field and selecting the app which will reveal a checkbox to resolve the field when using Contentful's GraphQL API. 
+
+### Using GraphQL app to see your app in action
+The simplest way test whether your app is resolving data correctly is to install the [GraphQL Playground app](https://app.contentful.com/deeplink?link=apps&id=graphql-playground). You should see a new field with a suffix `_data` which should contain data resolved from your app.
```
Comment (BasKiers) [inline examples/hosted-delivery-function-potterdb/INSTRUCTIONS.md:23]:
```suggestion
The simplest way test whether your app is resolving data correctly is to install the [GraphQL Playground app](https://app.contentful.com/deeplink?link=apps&id=graphql-playground). You should see a new field with a suffix `_data` when querying for your content type, which should contain data resolved from your app.
```

File: examples/hosted-delivery-function-potterdb/INSTRUCTIONS.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+## Building your Delivery Function
+
+### Creating an app
+You can create an app using CLI using `npm run create-app-definition`. This will prompt you to enter details for your new app and organization details. Make sure your app is enabled for `Entry field` option with both Short text and JSON Object selected in the supported field types.
```
[the remark this answers is not in the document]
Comment (kdamball) [reply examples/hosted-delivery-function-potterdb/INSTRUCTIONS.md:4]: There's two places you need to set a location. One when creating an app and assigning possible locations for it. The second is what you're referring to. 
Screenshot 2023-10-30 at 13 42 49

**Review by ghepting — APPROVED** [review]

File: examples/hosted-delivery-function-potterdb/INSTRUCTIONS.md (hunk 1 above)
Comment (kdamball) [reply examples/hosted-delivery-function-potterdb/INSTRUCTIONS.md:6]: > they could alternatively use the web-app to create a new link
Wdym by this?

**Review by ghepting — APPROVED** [review]

**Review by BasKiers — APPROVED** [review]

**Pull request merged**